_source/_posts/2020-01-21-angular-material-login.md

```diff
@@ -672,38 +683,42 @@ Finally, add the `login` route and the route guard to the router setup. Open `sr
 
 ```ts
 import { LoginComponent } from './login/login.component';
-import { AuthGuardService } from './auth-guard.service';
+import { AuthGuard } from './auth.guard';
 ```
 
 Now, change the `routes` array to look like the following.
 
 ```ts
 const routes: Routes = [
   {
-    path: '',
-    component: HomeComponent
+    path: 'game',
+    component: GameComponent,
+    canActivate: [ AuthGuard ]
   },
   {
     path: 'login',
     component: LoginComponent
   },
   {
-    path: 'game',
-    component: GameComponent,
-    canActivate: [ AuthGuardService ]
-  },
+    path: '',
+    component: HomeComponent
+  }
 ];
 ```
 
-Well done, that's it. The application is complete! 
+Well done, that's it. The application is complete!
 
 When you run the `ng serve` command and navigate to `http://localhost:4200` you should now be able to log in and out of the application. The login form uses Material Design and should look something like the image below.
 
 {% img blog/angular-material-login/finished-product.png alt:"Finished product" width:"800" %}{: .center-image }
 
+### Make Your App More Secure
+
+One thing you might notice is that we only watch for sign in and sign out actions to update the authenticated state. This works for our case, but you might not want to allow users whose sessions timed out to continue playing tic-tac-toe. To make your app more secure, you need to watch for sessions timing out. There's several ways to do this, but some popular ones include subscribing to `Router` events and checking, or if your app makes calls to a back-end, redirect the user to the login page when the HTTP call returns a 401 response code. 
+
 ### Make Angular Tests Pass With Angular Material
 
-You generated a lot of code in this tutorial. When you created components, tests were created for those components as well. The tests merely verify the components render. If you run `ng test`, most of them will fail because the tests don't have the imports for the components you added. If you'd like to see what it takes to make all the tests pass, see [this commit](https://github.com/oktadeveloper/okta-angular-material-login-example/commit/20e899d0ba4f8074681548e268337bd13153f140).
+You generated a lot of code in this tutorial. When you created components, tests were created for those components as well. The tests merely verify the components render. If you run `ng test`, most of them will fail because the tests don't have the imports for the components you added and a provider for the `OktaAuth` object. If you'd like to see what it takes to make all the tests pass, see [this commit to add the Angular Material library imports](https://github.com/oktadeveloper/okta-angular-material-login-example/commit/20e899d0ba4f8074681548e268337bd13153f140), and [this commit to provide a fake OktaAuth object](https://github.com/oktadev/okta-angular-material-login-example/commit/e06c1cc20e381b3a72f3c5597dc38dcd7095b82b).
```

Comment:
```suggestion
You generated a lot of code in this tutorial. When you created components, tests were created for those components as well. The tests merely verify that the components render. If you run `ng test`, most of them will fail because the tests have neither the imports for the components you added nor a provider for the `OktaAuth` object. If you'd like to see what it takes to make all the tests pass, look at [this commit to add the Angular Material library imports](https://github.com/oktadeveloper/okta-angular-material-login-example/commit/20e899d0ba4f8074681548e268337bd13153f140), and [this commit to provide a fake OktaAuth object](https://github.com/oktadev/okta-angular-material-login-example/commit/e06c1cc20e381b3a72f3c5597dc38dcd7095b82b).
```